app/workers/users/delete_worker.rb

```diff
@@ -1,15 +1,15 @@
 module Users
-  class SelfDeleteWorker
+  class DeleteWorker
     include Sidekiq::Worker
 
     sidekiq_options queue: :high_priority, retry: 10
 
-    def perform(user_id)
+    def perform(user_id, admin_delete = false)
```

Comment:
I generally don't like random boolean attributes, but kwargs don't play nicely with Sidekiq. Unless we want to notify the user in both cases, then I can get rid of this argument completely.